source/core_mqtt.c

```diff
@@ -2681,6 +2784,53 @@ MQTTStatus_t MQTT_InitStatefulQoS( MQTTContext_t * pContext,
 
 /*-----------------------------------------------------------*/
 
+MQTTStatus_t MQTT_InitRetransmits( MQTTContext_t * pContext,
+                                   MQTTStorePacketForRetransmit storeFunction,
+                                   MQTTRetrievePacketForRetransmit retrieveFunction,
+                                   MQTTClearPacketForRetransmit clearFunction,
+                                   MQTTClearAllPacketsForRetransmit clearAllFunction )
```

Comment:
Do we need clear all? or can we just call clear for all packetids we know?

Reply:
For a clean session, we do need to clear all the memory allocated to publish messages. It is simpler to ask the user to clear all the memory instead of looping through the array one entry at a time. 
But, on the other hand, it is an additional function that the user has to provide.

I can see the value in both approaches. But, I think that asking the user to provide one more function is bad for user experience. To that end, I think we can remove this.

Reply:
Thinking a bit more, having a clear all function seems useful. If the user is using an array - they can just memset that to 0.
Thoughts @archigup?